File: Day_19/Q75.java
import java.util.*;
class Q75{
    static Scanner sc = new Scanner(System.in);
    //Input The Matrix
    static void input_2D_array(int arr[][],int row,int col){
        for(int i = 0;i < row;i++){
            for(int j = 0;j < col;j++) arr[i][j] = sc.nextInt();
        }
    }
    //Display The Matrix
    static void display_2D_array(int arr[][],int row,int col){
        for(int i = 0;i < row;i++){
            for(int j = 0;j < col;j++) System.out.print(arr[i][j] + "\t");
            System.out.println();
        }
    }
    //Matrix Transpose
    static int[][] matrix_transpose(int matrix[][],int row,int col){
        int transposed_matrix[][] = new int[row][col];
        for(int i = 0;i < col;i++){
            for(int j = 0;j < row;j++){
                transposed_matrix[i][j] = matrix[j][i];
            }
        }
        return transposed_matrix;
    }
    public static void main(String[] args){
        int row = sc.nextInt();
        int col = sc.nextInt();
        //Input Validation
        if(row <= 0 || col <= 0){
            System.out.print("Invalid Input");
        }
        else{
            int matrix[][] = new int[row][col];
            int transposed_matrix[][] = new int[col][row];
            input_2D_array(matrix, row, col);
            transposed_matrix = matrix_transpose(matrix, row, col);
            display_2D_array(transposed_matrix, col, row);
        }
    }
}

File: Day_19/Q75.cpp
#include <bits/stdc++.h>
using namespace std;
//Input The 2D-Array
void input_array(vector<vector<int>> &arr,int rows,int col){
    for(int i = 0;i < rows;i++){
        for(int j = 0;j < col;j++) cin >> arr[i][j];
    }
}
//Display The 2D-Array
void display_matrices(vector<vector<int>> &arr,int rows,int col){
    for(int i = 0;i < rows;i++){
        for(int j = 0;j < col;j++) cout << arr[i][j] << "\t";
        cout << endl;
    }
}
//Transposing The Matrix
vector<vector<int>> transpose_matrix(vector<vector<int>> &matrix,int rows,int col){
    vector<vector<int>> transpose(col,vector<int>(rows));
    for(int i = 0;i < col;i++){
        for(int j = 0;j < rows;j++) transpose[i][j] = matrix[j][i];
    }
    return transpose;
}
int main(){
    int rows,col;
    cin >> rows >> col;
    //Input Validation
    if(rows <= 0 || col <= 0){
        cout << "Invalid Input";
        return 0;
    }
    vector<vector<int>> matrix(rows,vector<int>(col));
    //Performing Necessary Actions By Calling The Functions
    input_array(matrix,rows,col);
    vector<vector<int>> transposed_matrix = transpose_matrix(matrix,rows,col);
    display_matrices(transposed_matrix,col,rows);
    return 0;
}